reana_server/rest/workflows.py

```diff
@@ -413,13 +411,21 @@ def create_workflow(user):  # noqa
             workflow=workflow_dict, user=str(user.id_)
         ).result()
         if git_data:
-            Workflow.update_workflow_status(
-                Session, response["workflow_id"], RunStatus.queued
+            workflow = _get_workflow_with_uuid_or_name(
+                response["workflow_id"], str(user.id_)
             )
+            if workflow.type_ == "yadage":
+                _load_yadage_spec(workflow, workflow_dict["operational_options"])
+            complexity = _calculate_complexity(workflow)
+            total_cluster_memory = NodesStatus().get_total_memory()
+            workflow_priority = workflow.get_priority(total_cluster_memory)
+            workflow_min_job_memory = get_workflow_min_job_memory(complexity)
             current_workflow_submission_publisher.publish_workflow_submission(
                 user_id=str(user.id_),
-                workflow_id_or_name=response["workflow_id"],
+                workflow_id_or_name=workflow.get_full_workflow_name(),
                 parameters=request.json,
+                priority=workflow_priority,
+                min_job_memory=workflow_min_job_memory,
```

Comment:
Since this is repeated as well in `start_workflow`, maybe we can extract it and reuse it in both places?

It's a pity that we don't have [Flask Views](https://flask.palletsprojects.com/en/2.0.x/views/).. that way we could leverage inheritance to create reusable functions and keep everything organized. This would be an important step towards a model-view-controller architecture as we could extract business logic to different places and avoid having e.g `Session.commit()`, etc. in this layer. Just dumping ideas we might need to consider for the future. cc/ @tiborsimko @ParthS007 WDYT?